includes/class-amp-theme-support.php

```diff
@@ -540,6 +540,23 @@ public static function filter_the_content( $content ) {
 		return $sanitized_content;
 	}
 
+	/**
+	 * Determine the type of component script.
+	 *
+	 * @param string $component The component name.
+	 * @return string the type of component.
+	 */
+	public static function get_component_type( $component ) {
+		$component_types = apply_filters( 'amp_component_types', array(
+			'amp-mustache' => 'template',
+		) );
+
+		if ( ! empty( $component_types[ $component ] ) ) {
+			return $component_types[ $component ];
+		}
+		return 'element';
+	}
+
```

Comment:
@westonruter component scripts are have `custom-element` in the script tags, however `amp-mustache` used for `amp-list` does not, instead it has `custom-template`. see https://www.ampproject.org/docs/reference/components/amp-mustache
It looks like this is the only component that's different, however I separated it to its own method in case the spec changes, or additionals are added in future.

Reply:
I don't think get_component_type would be needed if we sanitize the entire body, right?

Reply:
@westonruter I think we still do, since this is done after the sanitisation and the scripts are placed in. All the scripts are `custom-element` but the template is `custom-template`.